lib/test_queue/runner.rb

```diff
@@ -334,9 +371,15 @@ def distribute_queue
       return if relay?
       remote_workers = 0
 
-      until @queue.empty? && remote_workers == 0
+      until @discovering_suites_pid.nil? && @queue.empty? && remote_workers == 0
         queue_status(@start_time, @queue.size, @workers.size, remote_workers)
 
+        # Make sure our discovery process is still doing OK.
+        if @discovering_suites_pid && Process.waitpid(@discovering_suites_pid, Process::WNOHANG) != nil
+          @discovering_suites_pid = nil
```

Comment:
To me they're the same question: "Are suites currently being discovered?" If the answer to that is "yes", it means two things:

1. the queue might get new items added to it even if it's currently empty
2. we have a subprocess we need to `wait` for so it doesn't become a zombie

I originally had another `@discovering_suites` boolean that was set to `true` when we forked the subprocess and `false` when the subprocess completed, but I removed it because it didn't feel like it added any clarity. You can see how it looks in https://github.com/tmm1/test-queue/commit/d3d3c572faff752572a36f25e74ac70183a521e4.